drivers/display/ssd1327.c

```diff
@@ -0,0 +1,396 @@
+/*
+ * Copyright (c) 2024 Savoir-faire Linux
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(ssd1327, CONFIG_DISPLAY_LOG_LEVEL);
+
+#include <string.h>
+#include <zephyr/device.h>
+#include <zephyr/init.h>
+#include <zephyr/drivers/display.h>
+#include <zephyr/drivers/gpio.h>
+#include <zephyr/drivers/spi.h>
+#include <zephyr/kernel.h>
+
+#include "ssd1327_regs.h"
+
+#define SSD1327_ENABLE_VDD				0x01
+#define SSD1327_ENABLE_SECOND_PRECHARGE			0x62
+#define SSD1327_VCOMH_VOLTAGE				0x0f
+#define SSD1327_PHASES_VALUE				0xf1
+#define SSD1327_DEFAULT_PRECHARGE_V			0x08
+#define SSD1327_UNLOCK_COMMAND				0x12
+
+union ssd1327_bus {
+	struct spi_dt_spec spi;
+};
+
+typedef bool (*ssd1327_bus_ready_fn)(const struct device *dev);
+typedef int (*ssd1327_write_bus_fn)(const struct device *dev, uint8_t *buf, size_t len,
+				    bool command);
+typedef const char *(*ssd1327_bus_name_fn)(const struct device *dev);
+
+struct ssd1327_config {
+	union ssd1327_bus bus;
+	struct gpio_dt_spec data_cmd;
+	struct gpio_dt_spec reset;
+	ssd1327_bus_ready_fn bus_ready;
+	ssd1327_write_bus_fn write_bus;
+	ssd1327_bus_name_fn bus_name;
+	uint16_t height;
+	uint16_t width;
+	uint8_t oscillator_freq;
+	uint8_t start_line;
+	uint8_t display_offset;
+	uint8_t multiplex_ratio;
+	uint8_t prechargep;
+	uint8_t remap_value;
+	bool color_inversion;
+};
+
+struct ssd1327_data {
+	uint8_t contrast;
+	uint8_t scan_mode;
+};
+
+#if (DT_HAS_COMPAT_ON_BUS_STATUS_OKAY(solomon_ssd1327fb, spi))
+static bool ssd1327_bus_ready_spi(const struct device *dev)
+{
+	const struct ssd1327_config *config = dev->config;
+
+	if (gpio_pin_configure_dt(&config->data_cmd, GPIO_OUTPUT_INACTIVE) < 0) {
+		return false;
+	}
+
+	return spi_is_ready_dt(&config->bus.spi);
+}
+
+static int ssd1327_write_bus_spi(const struct device *dev, uint8_t *buf, size_t len, bool command)
```

Comment:
This does need to use the [MIPI DBI](https://docs.zephyrproject.org/latest/hardware/peripherals/mipi_dbi.html) API, as @kartben highlighted. If you'd like an example to see how to convert this driver to that API, take a look here- https://github.com/zephyrproject-rtos/zephyr/pull/73946/files. Ideally I'd like to avoid the `write_bus` abstraction here- I understand that other SSD drivers do this, but if we instead use `mipi_dbi_command_write` and `mipi_dbi_display_write` these abstractions will serve a very similar purpose in the driver.